src/vm/util.hpp

```diff
@@ -804,16 +762,7 @@ if (CheckCanRunManagedCode && !CanRunManagedCode())     \
     return CannotEnterRetVal;                           \
 SO_INTOLERANT_CODE_NOTHROW(CURRENT_THREAD, return SORetVal)  \
 
-#define ComCallHostNotificationHR()                                         \
-ReverseEnterRuntimeHolderNoThrow REHolder;                                  \
-if (CLRTaskHosted())                                                        \
-{                                                                           \
-    HRESULT hr = REHolder.AcquireNoThrow();                                 \
-    if (FAILED(hr))                                                         \
-    {                                                                       \
-        return hr;                                                          \
-    }                                                                       \
-}
+#define ComCallHostNotificationHR() ReverseEnterRuntimeHolderNoThrow REHolder;
```

Comment:
ReverseEnterRuntimeHolder does nothing without the hosting. You can delete it from everywhere.